### ISSUE TITLE
Update CONTRIBUTING with detailed process for dev setup, branching, submitting PRs

### ISSUE THREAD

**Issue by jsheunis** [ME-ICA/tedana#346]
### Summary
New developers contributing to tedana should be able to follow an intuitive process for submitting a PR. To achieve this, CONTRIBUTING could contain a detailed process suggesting the steps a developer could follow, from the developer setup to opening an issue to submitting a PR, according to `tedana` best practices.

### Additional Detail
Currently in CONTRIBUTING, the [Making a change](https://github.com/ME-ICA/tedana/blob/master/CONTRIBUTING.md#making-a-change) section contains several steps instructing a developer how to go about creating issues, making changes and submitting a pull request. However, the developer setup required to do this is not immediately clear. Also, the preferred process for forking, branching, merging, etc, could be more intuitive for newcomers, e.g. a visual diagram or step-by-step explanation with accompanying git commands could help lower the barrier to entry. 

### Next Steps
- [ ] Tedana developers to share input on the required steps (Suggestion from @jbteves: fork -> clone fork -> branch from master -> open PR from branch -> merge PR -> update master -> delete branch)
- [ ] Create PR with suggested changes to CONTRIBUTING @jsheunis

**Comment by jbteves** [comment]
Pinging all @ME-ICA/tedana-devs; would be nice to have everyone's input.

**Comment by tsalo** [comment]
The recommended procedure is pretty common to using git or contributing on GitHub, and isn't specific to tedana. If possible, I'd like to point to an existing, very introductory walkthrough and explanation of the fork-branch-PR workflow. Of course most git-related texts end up being very confusing to beginners, so we'll need something that is aimed at beginners specifically.

**Comment by smoia** [comment]
@eurunuela might have a link or a document with what you're looking for, @tsalo !

**Comment by eurunuela** [comment]
You can check the slides I used for giving a lesson on git during BrainHack Donostia 2019. They are aimed at beginners and should cover the basics plus some tips. You chan check them here: https://git.bcbl.eu/brainhackdonostia/material-2019/blob/master/Git_GitLab_course.pdf

**Comment by jbteves** [comment]
Also, [GitHub itself](https://help.github.com/en/articles/fork-a-repo) has some tutorials specific to GitHub. They're quite nice, and I suspect a lot of people will be using the GitHub GUI as their git client.

**Comment by jsheunis** [comment]
I've also found [Github Guides](https://guides.github.com/) to be useful entry-level summaries, especially the one on [Understanding the GitHub flow](https://guides.github.com/introduction/flow/) and [Forking Projects](https://guides.github.com/activities/forking/).

**Comment by eurunuela** [comment]
> Also, [GitHub itself](https://help.github.com/en/articles/fork-a-repo) has some tutorials specific to GitHub. They're quite nice, and I suspect a lot of people will be using the GitHub GUI as their git client.

That was my main concern when I prepared the class. It's great you have found some tutorials for those who use the GUI.

**Comment by jsheunis** [comment]
## Update
I've made a minor update to the CONTRIBUTING file on a [new branch on my fork](https://github.com/jsheunis/tedana/blob/doc/update-contributing/CONTRIBUTING.md). This update mostly aims to make the developer instructions more intuitive from my perspective (having recently gone through the setup). I also added one sentence to the [installation instructions](https://github.com/jsheunis/tedana/blob/doc/update-contributing/README.md) in the README, to point developers the right way.

## Remarks
I feel like the **Making a change** steps might also need to include a step (in between making the changes and submitting the pull request) describing some testing procedures or standards. I assume the developer setup instructions sufficed previously, but I moved this earlier, since this felt more intuitive as it happens earlier in the process. So do you think we need some extra short guidelines on testing? Maybe something on how to render the readthedocs site locally? And/or a pointer to python code testing basics/standards?

## Requests
Please have a look at the suggested changes and remarks and let me know if you have any comments. I haven't submitted this as a PR yet, since I'm not sure whether it's in a close-to-final state yet and I don't know what the convention is for work that might still need several changes. If you'd rather I submit it as a WIP PR, please let me know.

**Comment by jbteves** [comment]
I would love to add some testing docs personally. I don't have a good step-by-step for you yet, but I know that at some point @tsalo and @emdupre had to walk me through it because I couldn't find any docs, and there's probably some (now-buried) Gitter messages about it.

**Comment by emdupre** [comment]
Thanks, @jsheunis ! I think this looks great. Would you mind opening an issue so your changes are easier to review ?

re testing docs: It should be as easy to re-create them locally as running `make html` from inside the `docs` folder ! Especially now that the build errors have been cleared up. Would be great to get that documented and to confirm that it works for you locally !

**Comment by jsheunis** [comment]
Sure, will submit a PR today.

With regards to testing, will add the docs testing suggestion to the PR. I think it will also make sense to link to the testing page referred to in #372 once it exists.

**Comment by jsheunis** [comment]
Note on testing `make html` for sphinx documentation:

This failed when I tried for the first few times, showing errors that some modules were not found. Before this, I had set up a new conda environment for tedana development using miniconda3. Then I ran `python setup.py develop`. But it seems that this had not installed all the required packages for some reason (including `sphinx-argparse`, `sphinx_rtd_theme` and `numpydoc`). I had to  `pip install` them separately and then everything worked.

A further question about `html make`: it looks like this generates the html files locally in the "docs/_build" folder, and it also changes some other files. These all relate to my local testing, so they shouldn't be part of the changes I push to my fork. Should I manually delete them before committing, or is this process automated in some way?
